Gems/Meshlets/Code/Source/Meshlets/MeshletsRenderObject.cpp

```diff
@@ -0,0 +1,800 @@
+            &m_meshletsData.meshlets[0], &m_meshletsData.meshlet_vertices[0], &m_meshletsData.meshlet_triangles[0],
+            &mesh.indices[0], mesh.indices.size(),
+            &mesh.vertices[0].px, mesh.vertices.size(), sizeof(GeneratorVertex),
+            max_vertices, max_triangles, cone_weight));
+            
+            if (!m_meshletsData.meshlets.size())
+            {
+                printf("Error generating meshlets - no meshlets were built\n");
+                return 0;
+            }
+
+            // this is an example of how to trim the vertex/triangle arrays when copying data out to GPU storage
+            const meshopt_Meshlet& last = m_meshletsData.meshlets.back();
+            m_meshletsData.meshlet_vertices.resize(last.vertex_offset + last.vertex_count);
+            m_meshletsData.meshlet_triangles.resize(last.triangle_offset + ((last.triangle_count * 3 + 3) & ~3));
+            uint32_t meshletsCount = (uint32_t) m_meshletsData.meshlets.size();
+            //----------------------------------------------------------------
+*/
+
+            ////////////////////////////
+            std::vector<meshopt_Meshlet> meshlets(max_meshlets);
+            std::vector<unsigned int> meshlet_vertices(max_meshlets * max_vertices);        // Vertex Index indirection map
+            std::vector<unsigned char> meshlet_triangles(max_meshlets * max_triangles * 3); // Meshlet triangles into the vertex index indirection - local to meshlet.
+
+            // NO scan seems to return more localized meshlets
+            meshlets.resize(meshopt_buildMeshlets(
+                &meshlets[0], &meshlet_vertices[0], &meshlet_triangles[0], &mesh.indices[0], mesh.indices.size(),
+                &mesh.vertices[0].px, mesh.vertices.size(), sizeof(GeneratorVertex), max_vertices, max_triangles, cone_weight));
+
+            // this is an example of how to trim the vertex/triangle arrays when copying data out to GPU storage
+            const meshopt_Meshlet& last = meshlets.back();
+            meshlet_vertices.resize(last.vertex_offset + last.vertex_count);
+            meshlet_triangles.resize(last.triangle_offset + ((last.triangle_count * 3 + 3) & ~3));
+            uint32_t meshletsCount = (uint32_t) meshlets.size();
+
+            m_meshletsData.Descriptors = meshlets;
+            m_meshletsData.IndicesIndirection = meshlet_vertices;
+            m_meshletsData.EncodeTrianglesData(meshlet_triangles);
+
+            // Some validation
+            m_meshletsData.ValidateData((uint32_t)meshlet_vertices.size());
+            std::vector<uint32_t> decodedIndices(meshlet_triangles.size());
+            m_meshletsData.GenerateDecodedIndices(decodedIndices);
+            ////////////////////////////
+
+            AZ_Warning("Meshlets", false, "Successfully generated [%d] meshlets\n", meshletsCount);
+            return meshletsCount;
+        }
+
+        uint32_t MeshletsRenderObject::CreateMeshlets(
```

Comment:
It looks like there are a couple instances of this method that do the same thing and could perhaps be DRYed up.

Reply:
+1, this looks the same as what's in MeshletsAssets.cpp. Can you take one or the other? And if you keep this one, refer to my comments on MeshletsAssets.cpp and apply them here. I'm skipping this function in the review since it appears to be the same.

Reply:
Guys, these two classes represent two different things - one is a meshlets asset that comes to demonstrate CPU side data representation of meshlets while the other is GPU implementation of the meshlets render object demonstrating a different approach and to to be mixed. 
They do no support each other and as per the documentation - one is used as the reference for the other, and each one should be stand alone.

Regarding the CreateMeshlets method - while I agree that one can have a different name, and yes, since one calls the other I can unite hence create a 300 lines method, doing that would be pretty bad programming in my book.
Notice that one is used to call the other as the data is not exposed right away, hence the separation, but including all will simply create a much too long method - not a good guideline.  
While I can call it somewhat a bit different, it seems redundant to change the name from CreateMeshlets to CreateMeshletsWithMesh and CreateMeshletsWithBuffers.
Can we resolve this one as 'nit' or does it matter to you enough to insist of a change (frankly I don't support uniting to one method).